fmriprep/cli/workflow.py

```diff
@@ -0,0 +1,165 @@
+    if missing:
+        build_log.critical(
+            "Cannot run fMRIPrep. Missing dependencies:%s",
+            '\n\t* %s'.join(["{} (Interface: {})".format(cmd, iface)
+                             for iface, cmd in missing])
+        )
+        retval['return_code'] = 127  # 127 == command not found.
+        return retval
+
+    config.to_filename(config_file)
+    build_log.info(
+        "fMRIPrep workflow graph with %d nodes built successfully.",
+        len(retval['workflow']._get_all_nodes())
+    )
+    retval['return_code'] = 0
+    return retval
+
+
+def build_boilerplate(config_file, workflow):
+    """Write boilerplate in an isolated process."""
+    from .. import config
+
+    config.load(config_file)
+    logs_path = config.execution.output_dir / 'fmriprep' / 'logs'
+    boilerplate = workflow.visit_desc()
+
+    if boilerplate:
+        citation_files = {
+            ext: logs_path / ('CITATION.%s' % ext)
+            for ext in ('bib', 'tex', 'md', 'html')
+        }
+        # To please git-annex users and also to guarantee consistency
+        # among different renderings of the same file, first remove any
+        # existing one
+        for citation_file in citation_files.values():
+            try:
+                citation_file.unlink()
+            except FileNotFoundError:
+                pass
+
+    citation_files['md'].write_text(boilerplate)
+
+    citation_files = {
+        ext: config.execution.output_dir / 'fmriprep' / 'logs'
+        / ('CITATION.%s' % ext) for ext in ('bib', 'tex', 'md', 'html')
+    }
+
+    if not config.execution.md_only_boilerplate and citation_files['md'].exists():
+        from subprocess import check_call, CalledProcessError, TimeoutExpired
+        from pkg_resources import resource_filename as pkgrf
+        from shutil import copyfile
+        # Generate HTML file resolving citations
+        cmd = ['pandoc', '-s', '--bibliography',
+               pkgrf('fmriprep', 'data/boilerplate.bib'),
+               '--filter', 'pandoc-citeproc',
+               '--metadata', 'pagetitle="fMRIPrep citation boilerplate"',
+               str(citation_files['md']),
+               '-o', str(citation_files['html'])]
+
+        config.loggers.cli.info(
+            'Generating an HTML version of the citation boilerplate...')
+        try:
+            check_call(cmd, timeout=10)
+        except (FileNotFoundError, CalledProcessError, TimeoutExpired):
+            config.loggers.cli.warning(
+                'Could not generate CITATION.html file:\n%s', ' '.join(cmd))
+
+        # Generate LaTex file resolving citations
+        cmd = ['pandoc', '-s', '--bibliography',
+               pkgrf('fmriprep', 'data/boilerplate.bib'),
+               '--natbib', str(citation_files['md']),
+               '-o', str(citation_files['tex'])]
+        config.loggers.cli.info(
+            'Generating a LaTeX version of the citation boilerplate...')
+        try:
+            check_call(cmd, timeout=10)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/poldracklab/fmriprep/pullRequest?prid=5112966)